language-support/ts/typedoc.bzl

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2020 The DAML Authors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@os_info//:os_info.bzl", "is_windows")
+
+def ts_docs(pkg_name):
+    "Macro for Typescript documentation generation with typedoc"
+
+    native.genrule(
+        name = "docs",
+        srcs = native.glob(["**/*.ts"], exclude = ["**/*test.ts"]) + [":README.md"],
+        tools = ["@language_support_ts_deps//typedoc/bin:typedoc"],
+        outs = [pkg_name + "-docs.tar.gz"],
+        cmd = """
+          # NOTE: we need the --ignoreCompilerErrors flag because we get errors when tsc is trying to
```

Comment:
That sounds a bit fragile. Does this mean that links to dependencies don’t work? At least in Haskell that’s commonly what happens if dependencies are weird in some way.

Reply:
It turns out this does break links to dependencies but fixing it is annoying so for now let’s keep it like this.

Reply:
Linking to other packages doesn't work, but I'm also not quite sure if it would work if  tsc could resolve the packages. I'm afraid, this is just as good as it gets right now. For testing we could work around this problem by running the tests on the javascript output, but jsdoc misses almost every function on the produced javascript.